### PR TITLE
chore: restore bot workflow to update attributions

### PR DESCRIPTION
**Pull request by cryptodev-2s**
## **Description**

This PR reinstates the workflow that was removed in [this PR](https://github.com/MetaMask/metamask-extension/pull/25180). Additionally, a safeguard has been added to the workflow to prevent it from running on a fork.

## **Related issues**

Fixes: [#1947](https://github.com/MetaMask/MetaMask-planning/issues/1947)

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by ustad56 — APPROVED** [review]

**Comment by cryptodev-2s** [comment]
waiting for security team review of all workflows

File: .github/workflows/update-attributions.yml [hunk 1]
```diff
@@ -0,0 +1,194 @@
+name: Update Attibutions
```
Comment (mcmire) [inline .github/workflows/update-attributions.yml:1]: Nit: typo

``` suggestion
name: Update Attributions

File: .github/workflows/update-attributions.yml [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+          PR_NUMBER: ${{ github.event.issue.number }}
+      - run: corepack enable
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+      - name: Install dependencies from cache
+        run: yarn --immutable --immutable-cache
+      - name: Generate Atributions
```
Comment (mcmire) [inline .github/workflows/update-attributions.yml:97]: Nit: typo

``` suggestion
      - name: Generate Attributions

File: .github/workflows/update-attributions.yml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        env:
+          HAS_CHANGES: ${{ steps.attributions-changes.outputs.HAS_CHANGES }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.issue.number }}
+
+  check-status:
+    name: Check whether the attributions update succeeded
+    runs-on: ubuntu-latest
+    needs:
+      - commit-updated-attributions
```
Comment (mcmire) [inline .github/workflows/update-attributions.yml:166]: I see that we are making the check for whether the repo is a fork for all jobs but this one, do we want to do this here as well?

**Review by mcmire — DISMISSED** [review]
A few suggestions but looks good regardless.

**Review by mcmire — DISMISSED** [review]
LGTM!

**Review by ustad56 — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -706,7 +706,9 @@
       "tsx>esbuild": false,
       "@metamask/eth-trezor-keyring>@trezor/connect-web>@trezor/connect>@trezor/protobuf>protobufjs": false,
       "firebase>@firebase/firestore>@grpc/proto-loader>protobufjs": false,
-      "@lavamoat/lavadome-core>@lavamoat/preinstall-always-fail": false
+      "@lavamoat/lavadome-core>@lavamoat/preinstall-always-fail": false,
+      "@metamask/desktop>ws>bufferutil": false,
```
Comment (Gudahtt) [inline package.json:710]: Were these changes included by mistake?

Comment (cryptodev-2s) [reply package.json:710]: @Gudahtt no the pipeline fails if they are not included

Comment (Gudahtt) [reply package.json:710]: But we aren't using the `@metamask/desktop` package anywhere

Comment (cryptodev-2s) [reply package.json]: @Gudahtt right make sense I just removed them

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by mcmire — APPROVED** [review]
LGTM as well.

**Pull request merged**